src/systems/ec/implementations/react/components/timeline/src/Timeline2.jsx

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import Button from '@ecl/ec-react-component-button';
+
+import TimelineItem from './Timeline2Item';
+
+const Timeline = ({
+  items,
+  toggleCollapsed,
+  toggleExpanded,
+  button,
+  className,
+  ...props
+}) => {
+  const classNames = classnames(className, 'ecl-timeline2');
+
+  // Calculate collapse threshold
+  const threshold =
```

Comment:
is this logic set in stone (i.e. always THE logic to follow)? should it be part of the component or should we create a helper for it?

Reply:
it is written in the specs

Reply:
I know it is written in the specs, but this logic doesn't seem "perfect". It already starts with "more than 13-14 items", it's not precise enough. "should make use". "if the show/hide is used". To me, it's more a guideline than a rule.